app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java

```diff
@@ -683,14 +758,277 @@ protected void onPostExecute(Void aVoid) {
+                int channelNumber2 = scienceLab.calculateDigitalChannel(arrayLists[0].get(1));
+                int channelNumber3 = scienceLab.calculateDigitalChannel(arrayLists[0].get(2));
+
+                digitalChannelArray.add(scienceLab.getDigitalChannel(channelNumber1));
+                digitalChannelArray.add(scienceLab.getDigitalChannel(channelNumber2));
+                digitalChannelArray.add(scienceLab.getDigitalChannel(channelNumber3));
+                edgeOption[0] = arrayLists[1].get(0);
+                edgeOption[1] = arrayLists[1].get(1);
+                edgeOption[2] = arrayLists[1].get(2);
+
+                ArrayList<Integer> modes = new ArrayList<>();
+                for (int i = 0; i < channelMode; i++) {
+                    switch (edgeOption[i]) {
+                        case "EVERY EDGE":
+                            digitalChannelArray.get(i).mode = EVERY_EDGE;
+                            modes.add(EVERY_EDGE);
+                            break;
+                        case "EVERY FALLING EDGE":
+                            digitalChannelArray.get(i).mode = EVERY_FALLING_EDGE;
+                            modes.add(EVERY_FALLING_EDGE);
+                            break;
+                        case "EVERY RISING EDGE":
+                            digitalChannelArray.get(i).mode = EVERY_RISING_EDGE;
+                            modes.add(EVERY_RISING_EDGE);
+                            break;
+                        case "EVERY FOURTH RISING EDGE":
+                            digitalChannelArray.get(i).mode = EVERY_FOURTH_RISING_EDGE;
+                            modes.add(EVERY_FOURTH_RISING_EDGE);
+                            break;
+                        case "DISABLED":
+                            digitalChannelArray.get(i).mode = DISABLED;
+                            modes.add(DISABLED);
+                            break;
+                        default:
+                            digitalChannelArray.get(i).mode = EVERY_EDGE;
+                            modes.add(EVERY_EDGE);
+                    }
+                }
+
+                scienceLab.startThreeChannelLA(modes, null, null);
+
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+
+                LinkedHashMap<String, Integer> data = scienceLab.getLAInitialStates();
+
+                try {
+                    Thread.sleep(500);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+
+                holder1 = scienceLab.fetchLAChannel(channelNumber1, data);
+
+                try {
+                    Thread.sleep(500);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+
+                holder2 = scienceLab.fetchLAChannel(channelNumber2, data);
+
+                try {
```

Comment:
@harsh-2711 can we create a method, say `Delay(long delay)` and use it in these places?

Reply:
Sure :+1: